File: BC-W4/PHP/Countable/Countable.php
<?php
    class Countabled {
        private static $totalCount = 0;
        
        public function __construct() {
            self::$totalCount += 1;
        }
        
        public function __destruct() {
            self::$totalCount -= 1;
        }
        
        public static function getTotalCount() {
            return self::$totalCount;
        }
    }
    
    $count1 = new Countabled();
    $count2 = new Countabled();
    $count4 = new Countabled();
    $count5 = new Countabled($count4);
    echo  Countabled::getTotalCount() . PHP_EOL;

?>

File: BC-W4/Static/Countable.cpp
#include <iostream>

class Countable {
    private: 
        static int totalCount;
        
    public:
        Countable() {
            totalCount += 1;
        }
        
        Countable(const Countable& copy) {
            totalCount += 1;
        }
        
        ~Countable() {
            totalCount -= 1;
        }
        
        Countable& operator=(const Countable& copy) {
            totalCount += 1;
        }
        
        static int getTotalCount();
};

int Countable::totalCount = 0;

int Countable::getTotalCount() {
    return totalCount;
}


int main() {
    Countable *count1 = new Countable();
    Countable *count2 = new Countable();
    Countable *count4 = new Countable(*count2);
    Countable *count3;
    *count3 = *count1;
    
    std::cout << Countable::getTotalCount() << std::endl;
    
    delete count1;
    
    std::cout << Countable::getTotalCount() << std::endl;
    
    return 0;
}
